src/libostree/ostree-bootloader-zipl.c

```diff
@@ -217,25 +221,38 @@ _ostree_secure_execution_generate_initrd (const gchar *initrd,
     if (glnx_regfile_copy_bytes (fd, out_initrd->fd, (off_t) -1) < 0)
       return glnx_throw_errno_prefix (error, "s390x SE: copying ramdisk");
   }
-
-  g_autofree gchar *tmpdir = g_mkdtemp (g_strdup ("/var/tmp/se-initramfs-XXXXXX"));
-
-  *out_initrdname = g_strdup_printf ("/proc/%d/fd/%d", getpid (), out_initrd->fd);
-  const char *const argv[] = {SECURE_EXECUTION_RAMDISK_TOOL, *out_initrdname, tmpdir, NULL};
-  g_autofree gchar *out = NULL;
-  g_autofree gchar *err = NULL;
-  int status = 0;
-  if (!g_spawn_sync (NULL, (char**)argv, NULL, G_SPAWN_SEARCH_PATH,
-                     NULL, NULL, &out, &err, &status, error))
-    return glnx_prefix_error(error, "s390x SE: spawning %s", SECURE_EXECUTION_RAMDISK_TOOL);
-
-  if (!g_spawn_check_exit_status (status, error))
+  #ifdef HAVE_LIBARCHIVE
+  // creating new cpio gzip archive with LUKS keys
+  g_autoptr(OtAutoArchiveWrite) a = archive_write_new ();
+  g_assert_cmpint (0, ==, archive_write_set_format_cpio_newc (a));
```

Comment:
I'd suggest turning this assertion (and all the ones below) into proper handled errors.

Reply:
agree, will do